server/src/main/kotlin/suwayomi/tachidesk/graphql/mutations/DownloadMutation.kt

```diff
@@ -0,0 +1,260 @@
+    )
+
+    fun deleteChapters(input: DeleteChapterDownloadsInput): DeleteChapterDownloadsPayload {
+        val (clientMutationId, chapters) = input
+
+        Chapter.deleteChapters(chapters)
+
+        return DeleteChapterDownloadsPayload(
+            clientMutationId = clientMutationId,
+            chapters = transaction {
+                ChapterTable.select { ChapterTable.id inList chapters }
+                    .map { ChapterType(it) }
+            }
+        )
+    }
+
+    data class DeleteChapterDownloadInput(
+        val clientMutationId: String? = null,
+        val id: Int
+    )
+    data class DeleteChapterDownloadPayload(
+        val clientMutationId: String?,
+        val chapters: ChapterType
+    )
+
+    fun deleteChapter(input: DeleteChapterDownloadInput): DeleteChapterDownloadPayload {
+        val (clientMutationId, chapter) = input
+
+        Chapter.deleteChapters(listOf(chapter))
+
+        return DeleteChapterDownloadPayload(
+            clientMutationId = clientMutationId,
+            chapters = transaction {
+                ChapterType(ChapterTable.select { ChapterTable.id eq chapter }.first())
+            }
+        )
+    }
+
+    data class EnqueueChapterDownloadsInput(
+        val clientMutationId: String? = null,
+        val ids: List<Int>
+    )
+    data class EnqueueChapterDownloadsPayload(
+        val clientMutationId: String?,
+        val downloadStatus: DownloadStatus
+    )
+
+    fun enqueueChapterDownloads(
+        input: EnqueueChapterDownloadsInput
+    ): CompletableFuture<EnqueueChapterDownloadsPayload> {
+        val (clientMutationId, chapters) = input
+
+        DownloadManager.enqueue(DownloadManager.EnqueueInput(chapters))
+
+        return future {
+            EnqueueChapterDownloadsPayload(
+                clientMutationId = clientMutationId,
+                downloadStatus = withTimeout(30.seconds) {
+                    DownloadStatus(DownloadManager.status.first { it.queue.any { it.chapter.id in chapters } })
+                }
+            )
+        }
+    }
+
+    data class EnqueueChapterDownloadInput(
+        val clientMutationId: String? = null,
+        val id: Int
+    )
+    data class EnqueueChapterDownloadPayload(
+        val clientMutationId: String?,
+        val downloadStatus: DownloadStatus
+    )
+
+    fun enqueueChapterDownload(
+        input: EnqueueChapterDownloadInput
+    ): CompletableFuture<EnqueueChapterDownloadPayload> {
+        val (clientMutationId, chapter) = input
+
+        DownloadManager.enqueue(DownloadManager.EnqueueInput(listOf(chapter)))
+
+        return future {
+            EnqueueChapterDownloadPayload(
+                clientMutationId = clientMutationId,
+                downloadStatus = withTimeout(30.seconds) {
+                    DownloadStatus(DownloadManager.status.first { it.queue.any { it.chapter.id == chapter } })
+                }
+            )
+        }
+    }
+
+    data class UnqueueChapterDownloadsInput(
```

Comment:
I saw this in the DownloadManager as well, but wouldn't it be `dequeue` isntead of `unqueue`?

Reply:
Thanks